### PR TITLE
Fix wrong C# code examples in making plugins tutorial

### PR DESCRIPTION
**Pull request by mateuseap**
## What I did

I made adjustments to several C# code examples in the `making_plugins.rst` tutorial, located at the following path: `tutorials/plugins/editor/making_plugins.rst`. These changes were suggested and discussed by @raulsntos in the following comment: https://github.com/godotengine/godot-docs/pull/7867#discussion_r1312975013.

### CODE REVIEW

File: tutorials/plugins/editor/making_plugins.rst [hunk 1]
```diff
@@ -367,21 +367,21 @@ The script could look like this:
...
     public partial class CustomDock : EditorPlugin
     {
-        Control dock;
+        private Control _dock;
 
         public override void _EnterTree()
         {
-            dock = (Control)GD.Load<PackedScene>("addons/my_custom_dock/my_dock.tscn").Instantiate();
-            AddControlToDock(DockSlot.LeftUl, dock);
+            _dock = (Control)GD.Load<PackedScene>("res://addons/MyCustomDock/MyDock.tscn").Instantiate();
```
Comment (raulsntos) [inline tutorials/plugins/editor/making_plugins.rst:374]: nit: I think we prefer using the generic method instead, but I think it's fine either way.

```suggestion
            _dock = GD.Load<PackedScene>("res://addons/MyCustomDock/MyDock.tscn").Instantiate<Control>();
```

**Review by raulsntos — APPROVED** [review]
Thanks, LGTM.

File: tutorials/plugins/editor/making_plugins.rst (hunk 1 above)
Comment (mateuseap) [reply tutorials/plugins/editor/making_plugins.rst]: Done!

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you! Merged.

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.1

**Pull request merged**